### PR TITLE
EFSOI additions to util/EFSOI-Utilities/scripts directory for issue #118

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
The following files in util/EFSOI-Utilities/scripts are added for reading and visualizing the contents of osense files

	new file:   README_fv3gfs
	new file:   convdata_codes.csv
	new file:   copyefsoifiles.sh
	new file:   display_osense.py
	new file:   getefsiofiles.sh
	new file:   osense.py
	new file:   read_osense.py

The following are for unit testing the EFSOI task:

	new file:   unit_test/README
	new file:   unit_test/expdir/config.anal
	new file:   unit_test/expdir/config.base
	new file:   unit_test/expdir/config.efsoi
	new file:   unit_test/expdir/config.nsst
	new file:   unit_test/expdir/config.resources
	new file:   unit_test/jobscript.sh

The following scripts are no longer useful for the current versions of GFS/GDAS/EFSOI:

	deleted:    efso_advance_ensemble.sh
	deleted:    efso_cleanup.sh
	deleted:    efso_cleanup_wrapper.sh
	deleted:    efso_file_assembly.sh
	deleted:    efso_run.sh
	deleted:    efso_wrapper.sh

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
tagging @aerorahul @CatherineThomas-NOAA @MichaelLueken-NOAA (Liaofan not taggable)

**Comment by AndrewEichmann-NOAA** [comment]
unit test scripts removed

**Review by aerorahul — COMMENTED** [review]
See comments.
Rest are ok.

File: util/EFSOI_Utilities/scripts/copyefsoifiles.sh [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#!/bin/sh -x
```
[the remark this answers is not in the document]
Comment (AndrewEichmann-NOAA) [reply util/EFSOI_Utilities/scripts/copyefsoifiles.sh]: Script removed

File: util/EFSOI_Utilities/scripts/copyefsoifiles.sh [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+################################################################################
+
+
+
+COMROT=
+STGDIR=
+
+EDATE=2019111818
+NENS=20
+NDATE=/scratch1/NCEPDEV/global/Fanglin.Yang/save/VRFY/vsdb/nwprod/util/exec/ndate
```
[the remark this answers is not in the document]
Comment (AndrewEichmann-NOAA) [reply util/EFSOI_Utilities/scripts/copyefsoifiles.sh]: script removed

File: util/EFSOI_Utilities/scripts/getefsiofiles.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
+#!/bin/sh -x
```
[the remark this answers is not in the document]
Comment (AndrewEichmann-NOAA) [reply util/EFSOI_Utilities/scripts/getefsiofiles.sh]: script removed

**Review by CatherineThomas-NOAA — APPROVED** [review]
Thanks for addressing Rahul's and my comments.  Looks good to me.

**Review by aerorahul — APPROVED** [review]
looks good.
Thanks for addressing the comments.

**Comment by MichaelLueken-NOAA** [comment]
@AndrewEichmann-NOAA Similar to PR #146, please reduce the number of commits to just one using the following commands (hopefully the inclusion of an additional step will remove the error you encountered while attempting this for PR #146):

1) Clone a temporary copy of your fork

     git clone --recursive git@github.com:AndrewEichmann-NOAA/GSI.git update

2) Check out your EXP-efso_fv3_PR_util_EFSOI_scripts branch

     git checkout EXP-efso_fv3_PR_util_EFSOI_scripts

3) Rebase the NOAA-EMC/GSI master into your PR branch

```
     git remote add upstream git@github.com:NOAA-EMC/GSI.git
     git remote update
     git rebase upstream/master
     vi CMakeLists.txt
     Please remove lines 263 and 276 - 287.
     git add CMakeLists.txt
     git rebase --continue
     vi CMakeLists.txt
     Please remove lines 263 and 275 - 282.
     git add CMakeLists.txt
     git rebase --continue

```
4) Clean up commit messages

```
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
     git reset --soft HEAD~1
```

5) Commit and push back to the repository

```
     git commit -m "GitHub Issue #118. Implement EFSOI process in FV3 GFS workflow - EFSOI additions to util/EFSOI-Utilities/scripts directory."
     git push origin EXP-efso_fv3_PR_util_EFSOI_scripts --force
```

If you encounter the same issue as you did with PR #146, please let me know.

**Comment by AndrewEichmann-NOAA** [comment]
@MichaelLueken-NOAA Done. No errors this time.

**Comment by MichaelLueken-NOAA** [comment]
Both @aerorahul and @CatherineThomas-NOAA have reviewed these changes and have granted approval.  The regression tests were run and all passed successfully.  Since there were no changes to source code files, I will now grant final approval and merge this PR into the authoritative repository.

**Pull request merged**